### PR TITLE
Add examples for f407 and f411 ST dev boards

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Now F4xx boards are starting to pop up, I have added some discovery boards to `handwired/onekey` as starting points, and general feature testing.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
I'm guessing this is going to break if I merge, considering the ChibiOS upgrade has happened since.
I'll sort out a PR for your branch to bring things up to par, so we can get this one sorted.

File: keyboards/handwired/onekey/rules.mk [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
-BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
-UNICODE_ENABLE = no         # Unicode
+MIDI_ENABLE = no            # MIDI support
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (fauxpark) [inline keyboards/handwired/onekey/rules.mk:18]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/handwired/onekey/rules.mk [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
-HD44780_ENABLE = no         # Enable support for HD44780 based LCDs (+400)
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/handwired/onekey/rules.mk:19]:
```suggestion
```

File: keyboards/handwired/onekey/stm32f407g_disc1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
+## chip/board settings
```
Comment (fauxpark) [inline keyboards/handwired/onekey/stm32f407g_disc1/rules.mk:1]: Perhaps we can add these to mcu_selection.mk while we're at it?

**Pull request closed without merging**